core/unix/coredump.c

```diff
@@ -533,8 +585,8 @@ os_dump_core_internal(dcontext_t *dcontext)
         return false;
     }
     // Write section headers to the core dump file.
-    // TODO i#7046: Handle multiple program headers.
-    ELF_OFF file_offset = sizeof(ELF_HEADER_TYPE) + PROGRAM_HEADER_LENGTH;
+    file_offset = sizeof(ELF_HEADER_TYPE) +
+        sizeof(ELF_PROGRAM_HEADER_TYPE) * section_count + PROGRAM_HEADER_NOTE_LENGTH;
```

Comment:
I'm convinced now that a new variable program_header_count should be added to avoid confusion and errors in all these places that are using section_count.

Reply:
Good point. I've added a new variable program_header_count for the number of program headers.